pilot/pkg/model/push_context.go

```diff
@@ -1667,7 +1677,9 @@ func (ps *PushContext) SetDestinationRules(configs []config.Config) {
 		// We only honor . and *
 		if len(rule.ExportTo) == 0 && ps.exportToDefaults.destinationRule[visibility.Private] {
 			isPrivateOnly = true
-		} else if len(rule.ExportTo) == 1 && exportToMap[visibility.Private] {
+		} else if len(rule.ExportTo) == 1 && (exportToMap[visibility.Private]) {
+			isPrivateOnly = true
+		} else if rule.GetWorkloadSelector() != nil {
```

Comment:
ditto

Reply:
Same comment as above, I added this as a double check, even though we validate things beforehand.